Add a new section on how to collect WAF logs using Amazon Data Firehose

alaudazzi opened the pull request:
This PR adds a new Firehose use case on WAF logs. It reuses some of this [content](https://github.com/zmoog/public-notes/issues/83), kindly drafted by Maurizio Branca (thank you!).   

Doc [preview](https://observability-docs_bk_3850.docs-preview.app.elstc.co/guide/en/observability/master/monitor-aws-waf-firehose.html).
Closes https://github.com/elastic/observability-docs/issues/3837

alaudazzi commented:
@zmoog 
While testing the tutorial, I found a few points where the procedure might require additional clarification:

=== **Step 1** ===
I updated and simplified the instructions to reflect what actually appears on the UI. I've also removed the note saying: **"Firehose integration is currently in beta. Make sure to enable Display beta integrations."** because Amazon Data Firehose appear _in technical preview_, whether the option **Display beta integrations** is enabled or not.

=== **Step 2** ===
This shouldn't be a step, it's more of an introduction to what the WAF-compatible resources are. It makes more sense to move this info to step 5 where we give instructions on how to *Describe web ACL and associate it to an AWS resource*.

=== **Step 4** ===
1. Let's clarify better how to use these parameters ** `es_datastream_name` and `include_cw_extracted_fields`
2. Before you can move to the next steps, in the Backup settings you are prompted to specify an S3 bucket. If you don't have an existing one, you need to create a new one.

zmoog commented:
> === Step 2 ===
> This shouldn't be a step, it's more of an introduction to what the WAF-compatible resources are.

Yeah, I get what you mean. 

This is another instance of the problem we discussed offline regarding the AWS Network Firewall. Users may already have a resource ready to go, or they may be testing and don't have one, or they may want to run it on some dev/test environment.

We can probably solve this issue by assuming that users already have a target resource at hand. The "cookbook" repo with tools to create on-click test resources would be a great addition, but we can work on it as a separate work item.

zmoog commented:
> === Step 4 ===
> 
> Let's clarify better how to use these parameters ** es_datastream_name and include_cw_extracted_fields
> Before you can move to the next steps, in the Backup settings you are prompted to specify an S3 bucket. If you don't have an existing one, you need to create a new one.

Okay. I can work on these ones.

zmoog commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,141 @@
+
+. In {kib}, navigate to *Management* > *Integrations* and browse the catalog to find the Amazon Data Firehose integration.
+
+. Navigate to the *Settings* tab and click *Install Amazon Data Firehose assets*.
```
The instructions in this guide require the AWS integration (we are using the `logs-aws.waf-default` data stream)

Since the Amazon Data Firehose integration setup also requires AWS integration, which one should I mention here?

![CleanShot 2024-05-07 at 16 21 44](https://github.com/elastic/observability-docs/assets/25941/115672ad-e876-4630-895e-eac683d9f7ee)

alaudazzi replied:
@zmoog  sure I get the point.... let's review this on Monday

zmoog commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,141 @@
+
+. Click *Create Firehose stream* and choose the source and destination of your Firehose stream. Unless you are streaming data from Kinesis Data Streams, set source to `Direct PUT` and destination to `Elastic`. 
+
+. Provide a meaningful *Firehose stream name* that will allow you to identify this delivery stream later.
```
WAF restricts Firehose stream names. The warning "Your Firehose name must start with the prefix `aws-waf-logs—`, or it will not show up later." should go here.

zmoog commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,141 @@
++
+* *Parameters*:
++
+  ** `es_datastream_name`: Your Firehose name must start with the prefix `aws-waf-logs-` or it will not show up later.
+  ** `include_cw_extracted_fields`: This parameter is optional and can be set when using a CloudWatch logs subscription filter as the Firehose data source. When set to true, extracted fields generated by the filter pattern in the subscription filter will be collected. Setting this parameter can add many fields into each record and may significantly increase data volume in Elasticsearch. As such,  ...
```
The suggested `es_datastream_name` value should be `logs-aws.waf-default`. 

IIRC, WAF logs don't need the `include_cw_extracted_fields` for the basic use case.

alaudazzi replied:
I'll remove the `include_cw_extracted_fields` parameter

alaudazzi commented:
@zmoog Thank you for your comments!
I integrated almost all of them. After we finalize [step 4](https://github.com/elastic/observability-docs/pull/3850/#issuecomment-2095698015) we should be good to merge I guess. Let's chat more on Monday.

alaudazzi commented:
@zmoog 
as discussed this morning, here the last fixes:
- moved the WAF-compatible resources to the step "Create a web ACL" (RE this [comment](https://github.com/elastic/observability-docs/pull/3850#issuecomment-2095693204))
- adjusted step numbering
- cleaned up the list of parameters and clarified how to use `es_datastream_name` (RE this [comment](https://github.com/elastic/observability-docs/pull/3850#issuecomment-2095698015))
- Reworded the Kibana settings to avoid confusion with Amazon Data Firehose integration vs AWS integration (see this [comment](https://github.com/elastic/observability-docs/pull/3850#discussion_r1592583599))

If you can have a final check, we should be good to merge.

bmorelli25 commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,128 @@
+
+[discrete]
+[[firehose-waf-step-seven]]
+== Step 7: Visualize your WAF logs in {kib}
```
We skip from `Step 5` to `Step 7` (there's no `Step 6`).

bmorelli25 commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,128 @@
+
+[discrete]
+[[firehose-waf-step-one]]
+== Step 1: Install AWS integration in {kib}
```
```suggestion
== Step 1: Install the AWS integration in {kib}
```

zmoog commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,128 @@
++
+* *Retry duration*: Determines how long Firehose continues retrying the request in the event of an error. A duration of 60-300s should be suitable for most use cases.
++
+* *es_datastream_name*: Your Firehose name must start with the prefix `aws-waf-logs-` or it will not show up later.
```
The `aws-waf-logs-*` prefix restriction applies to the _stream name_ only. The `es_datastream_name` parameter value should be `logs-aws.waf-default`.

```suggestion
* *es_datastream_name*: `logs-aws.waf-default`
```

zmoog commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,128 @@
+
+[discrete]
+[[firehose-waf-step-four]]
+== Step 4: Create a web ACL
+
+To create a new web ACL, follow these steps:
```
I used "web ACL" mainly because AWS uses it as a noun in the documentation and the web console.

![CleanShot 2024-05-14 at 15 28 41@2x](https://github.com/elastic/observability-docs/assets/25941/6d043005-ba6a-4a71-a5b0-d606ce0bdfe4)

zmoog commented on docs/en/observability/cloud-monitoring/aws/monitor-aws-waf-firehose.asciidoc:
```diff
@@ -0,0 +1,128 @@
+- *Logging destination*: select "Amazon Data Firehose stream"
+- *Amazon Data Firehose stream*: select the Firehose stream you created in step 3.
+
+WAF creates the required IAM role.
```
Like the "web ACL" thing, the "IAM role" feels like a noun.

As an AWS user, I immediately identify what a writer means saying "IAM user", "IAM policy", or "IAM role" — "a user|role|policy from the IAM service in AWS":

![CleanShot 2024-05-14 at 15 38 31@2x](https://github.com/elastic/observability-docs/assets/25941/1ea3347f-61bb-4beb-bcb0-78f93e00e38b)

However, "web ACL" and "IAM role" work both ways.

alaudazzi commented on docs/en/observability/cloud-monitoring/aws/monitor-amazon-intro.asciidoc:
```diff
@@ -38,6 +38,8 @@ include::monitor-aws-vpc-flow-logs.asciidoc[leveloffset=+2]
 
 include::monitor-aws-cloudtrail-firehose.asciidoc[leveloffset=+2]
 
+include::monitor-aws-waf-firehose.asciidoc[leveloffset=+2]
```
Which new docs and which section? Sorry, I'm confused...

bmorelli25 replied:
Ignore this.
